source/slang/slang-options.cpp

```diff
@@ -531,6 +531,10 @@ struct OptionsParser
                 {
                     requestImpl->getFrontEndReq()->shouldDumpAST = true;
                 }
+                else if (argStr == "-doc")
+                {
+                    requestImpl->getFrontEndReq()->shouldDocument = true;
+                }
```

Comment:
This is probably fine/reasonable for now, but it would be really nice in the long run if `slang-doc` where a different tool than `slangc`. I guess it creates the problem that they would end up needing/wanting a lot of the same options, though, so yeah, it seems best to not tackle that just yet.